### ISSUE TITLE
Normalizacja wejścia sieci — dodać LayerNorm przed Conv1D

### ISSUE THREAD

**Issue by ArgonZapan** [ArgonZapan/DQN-transformer#29]
## Problem

8 cech wejściowych trafia do sieci z fundamentalnie różnymi skalami:

| Cecha | Zakres typowy |
|-------|--------------|
| `normalizedClose` | −3 ... +3 (z-score) |
| `normalizedVolume` | 0 ... 100+ (spike volume) |
| `rsiNorm` | 0 ... 1 |
| `macdNorm` | −∞ ... +∞ (niesklipowany) |
| `relativeRange` | 0 ... 0.05 |
| `candleDirection` | −0.1 ... +0.1 |
| `pctChange` | −0.1 ... +0.1 |
| `aboveSma` | 0 lub 1 |

Sieć musi sama nauczyć się ignorować różnice skal — utrudnia to trening i powoduje dominację cech o dużych wartościach (np. `normalizedVolume`, `macdNorm`) w gradiencie.

---

## Rozwiązanie — LayerNorm na wejściu sieci

Dodać `nn.LayerNorm(8)` jako pierwszą warstwę w sieci, **przed** blokami Conv1D. LayerNorm normalizuje każdą świecę po wymiarze cech (po 8 wartościach), wyrównując skale bez ręcznej kalibracji progów.

Dla każdego wektora `[f1, ..., f8]`:
```
output[i] = γ[i] * (fi − μ) / (σ + ε) + β[i]
```
gdzie `γ` i `β` to **wyuczone parametry** (16 liczb łącznie) — sieć sama decyduje jak skalować każdą cechę.

### Zmiana w `python/model/network.py`

```python
class DQNNetwork(nn.Module):
    def __init__(self, ...):
        # ... istniejący kod ...
        self.input_norm = nn.LayerNorm(8)   # ← nowe, normalizuje po wymiarze cech

    def forward(self, state_tensors, action_mask=None):
        # Zastosuj LayerNorm przed Conv1D
        state_tensors = [self.input_norm(x) for x in state_tensors]
        # ... reszta forward() bez zmian ...
```

`state_tensors` ma kształt `[batch, seq_len, 8]` — LayerNorm działa po ostatnim wymiarze (8 cech), co jest domyślnym zachowaniem `nn.LayerNorm(8)`.

---

## Dlaczego LayerNorm, nie BatchNorm

BatchNorm normalizuje po całym batchu i wymaga stabilnych statystyk rozkładu danych. W RL rozkład danych zmienia się w trakcie treningu (wraz z polityką agenta) — BatchNorm jest więc problematyczny. LayerNorm operuje per-próbka i nie ma tego problemu.

---

## Zalety

- **Zero zmian w Node.js** — normalizacja wejścia przeniesiona do sieci
- **Adaptuje się automatycznie** — działa poprawnie na nowych danych i różnych rynkach bez rekalibracji
- **Koszt:** +16 parametrów (γ i β per cecha) — pomijalny
- **Eliminuje dominację** cech o dużych wartościach (`normalizedVolume`, `macdNorm`) w gradiencie
